presto-kudu/src/main/java/io/prestosql/plugin/kudu/KuduConnector.java

```diff
@@ -45,6 +46,8 @@
     private final KuduTableProperties tableProperties;
     private final ConnectorPageSinkProvider pageSinkProvider;
     private final Set<Procedure> procedures;
+    private final ConnectorNodePartitioningProvider nodePartitioningProvider;
+    private final KuduSessionProperties sessionProperties;
```

Comment:
Well, should I bound List<PropertyMetadata<?>> sessionProperties in KuduMoudle or somewhere else? Because after I change KuduSessionProperties to List<PropertyMetadata<?>> it shows error like below:
`1) Explicit bindings are required and java.util.List<io.prestosql.spi.session.PropertyMetadata<?>> is not explicitly bound.
  while locating java.util.List<io.prestosql.spi.session.PropertyMetadata<?>>
    for the 9th parameter of io.prestosql.plugin.kudu.KuduConnector.<init>(KuduConnector.java:65)
  at io.prestosql.plugin.kudu.KuduModule.configure(KuduModule.java:56)
`

Reply:
Agree with you, I have changed my code according to CassandraConnector.  :)

Reply:
> A few minor changes. How do we handle for multi level hash based partitioning ? If we partition tables on based on two hash partition, how can we handle them ?

I think we can check KuduBucketFunction#getBucket, in this method we judge which bucket one record belongs to by all its bucket columns(see param bucketChannelTypes of method getBucketId). So I think it won't be a problem when we handling multi level hash based partitioning.   :)